lib/blemulator/internal/bridge/platform_to_dart_bridge.dart

```diff
@@ -3,13 +3,57 @@ part of internal;
 class PlatformToDartBridge {
   SimulationManager _manager;
   MethodChannel _platformToDartChannel;
+  Map<String, CancelableOperation> pendingTransactions = HashMap();
 
   PlatformToDartBridge(this._manager) {
     _platformToDartChannel = new MethodChannel(ChannelName.platformToDart);
     _platformToDartChannel.setMethodCallHandler(_handleCall);
   }
 
-  Future<dynamic> _handleCall(MethodCall call) {
+  Future<dynamic> _handleCall(MethodCall call) async {
+    String transactionId = call.arguments[SimulationArgumentName.transactionId];
+    Future<dynamic> platformCallResultFuture = _dispatchPlatformCall(call);
```

Comment:
You're not checking if the called operation even _is_ cancellable, you always add it the map of pending transactions with null key (since non-cancellables won't have `transactionId`).